modules/persistenceService/src/main/scala/subsPersistenceLayer/SubscriptionServicePersistenceLayer.scala

```diff
@@ -43,10 +54,6 @@ object SubscriptionServiceQuery {
     val sql = "insert into subscriptions(user_id, repository_id) values(?,?)"
     Update[SubscriptionInfo](sql)
   }
-
-  def delete(id: Id): Update[PostSubscriptionData] = {
-    val sql =
-      s"delete from subscriptions where id = ${id.id} and repository_id in (select * from repositories where owner = ? and repository = ?)"
-    Update[PostSubscriptionData](sql)
-  }
+  def delete(id: Id, organization: Owner, repository: Repository): Update0 =
+    sql"delete from subscriptions where user_id = ${id.id} and repository_id in (select repository_id from repositories where owner=${organization.owner} and repository=${repository.repository})".update
```

Comment:
Minor thing, but I think we should have `Meta` for the wrapper types and then use the type directly here:
```suggestion
    sql"delete from subscriptions where user_id = $id and repository_id in (select repository_id from repositories where owner=$organization and repository=$repository)".update
```
Bear in mind that we have to specify the inner field here, but that field could be named differently. i.e.

```scala
final case class Owner(owner: String) extends AnyVal
// vs 
final case class Owner(s: String) extends AnyVal
// vs 
final case class Owner(value: String) extends AnyVal
```

Reply:
sweet, I will make the changes and user Meta.
Thanks @fedefernandez 🙂

Reply:
shouldn't they be automatically generated?

Reply:
> shouldn't they be automatically generated?

what do you mean by this please @moleike ?

Reply:
I mean, if you import doobie.implicits._ I think you will have (shapeless-based?) instances in scope for your value classes. I am not sure about the mechanics of this. This works for me
```scala
import doobie.implicits._

  def deleteSubscriptionsQ(user: UserId) = {
    val sql = s"""
      DELETE FROM subscriptions AS s
      USING repositories AS r
      WHERE user_id='${user}'
      AND r.owner=? AND r.repository=?
      AND r.repository_id = s.repository_id"""

    Update[Repository](sql)
  }
```

Reply:
If the concept of instance derivation is confusing, it's normal 😄. But the idea is that most types have *boring* instances (just pure repetition). There are many ways to solve this problem, but one that I think used to be popular is to use datatype-generic programming libraries like shapeless in Scala.

Reply:
Sweet, thank you very much @moleike 🙂

Reply:
Now I need to find out what `shapeless-based` instances are 😅
So much to learn.